modin/backends/pandas/query_compiler.py

```diff
@@ -1862,15 +1862,68 @@ def mode_builder(df):
     def fillna(self, **kwargs):
         """Replaces NaN values with the method provided.
 
+        Parameters
+        ----------
+            squeeze_self : boolean
+                The flag to squeeze self.
+            squeeze_value : boolean
+                The flag to squeeze value (this flag is applied if value is query compiler and denotes a Series type).
+
         Returns:
             A new QueryCompiler with null values filled.
         """
+        squeeze_self = kwargs.pop("squeeze_self", False)
+        squeeze_value = kwargs.pop("squeeze_value", False)
         axis = kwargs.get("axis", 0)
         value = kwargs.get("value")
         method = kwargs.get("method", None)
         limit = kwargs.get("limit", None)
         full_axis = method is not None or limit is not None
-        if isinstance(value, dict):
+        if isinstance(value, BaseQueryCompiler):
+            kwargs.pop("value")
+            if squeeze_self:
+                if full_axis:
+                    value = value.to_pandas().squeeze(axis=1)
+
+                    def fillna_builder(series):
+                        # Limit parameter works only on Series type, so we have to squeeze both objects to get
+                        # correct behavior.
+                        return series.squeeze(axis=1).fillna(value=value, **kwargs)
+
+                    new_modin_frame = self._modin_frame._apply_full_axis(
```

Comment:
What is better?
1 - when we turn `value` into pandas and call `_apply_full_axis`
2 - when we don't turn `value` into pandas at once and call `broadcast_apply_full_axis`.

Reply:
Both approaches look good to me but I am concerned about the performance of each one. Can you check it, please?

Reply:
I ran ASV benchmarks that we have for Fillna and looks like the current (`apply_full_axis`) approach is better than modified (`broadcast_apply_full_axis`). Here are relevant measurements for current approach:

                 Series      Series      (5000, 5000)     282±0ms        266±0ms       590±0ms      277±0ms   
                 Series      Series     (1000000, 10)     68.5±0ms       98.4±0ms      53.1±0s      136±0ms   

and for modified:

                 Series      Series      (5000, 5000)     282±0ms        309±0ms       585±0ms      296±0ms   
                 Series      Series     (1000000, 10)     107±0ms        104±0ms       52.1±0s      164±0ms

Reply:
Okay, let's keep as it is. Thanks!